support-frontend/app/services/stepfunctions/Client.scala

```diff
@@ -27,19 +29,32 @@ object Client {
 
     new Client(client, arn)
   }
+
+  def generateExecutionName(name: String, l: Long = System.nanoTime()) = {
+    val uniq = {
+      val bytes = ByteBuffer.allocate(8).putLong(l).array()
+      val encodedString = Base64.getEncoder.encodeToString(bytes)
+      encodedString.replaceAll("\\+", "_").replaceAll("/", "_").replaceAll("=", "")
+    }
```

Comment:
I think I can see what's potting here, is it a base64 encoded string to an 8bytes string?

Reply:
Ultimately it's turning a 64 bit long into 11 characters that are valid in an execution name.

1. turn the Long (64 bits = 8 x 8bit byte) into 8 bytes in an array.
2. encode the byte array into base64 (6 bits per character - 64 bits makes 11 characters)
3. base64 includes 0-9, a-z, A-Z, `+` and `/` i.e. 10 + 26 + 26 + 2 = 64.  Our extra characters are `-` and `_` so it's just a quick substitution.  The `=` is just padding as base64 always pads to a multiple of 4 characters i.e. 3 bytes, so.we just throw that away.